### PR TITLE
More correct first place recalculation on unban

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: common/ripple/user_utils.py [hunk 1]
```diff
@@ -951,36 +951,35 @@ async def recalculate_and_update_first_place_scores(user_id: int) -> None:
...
-                    order,
-                    table_name,
-                ),
-                [score["beatmap_md5"], score["play_mode"]],
+            existing_first_place = await glob.db.fetch(
+                f"""
+                SELECT scores_first.scoreid, scores_first.userid, scores.{order} AS score_value
+                FROM scores_first
+                WHERE beatmap_md5 = %s
+                JOIN {table_name} AS scores ON scores.id = scoreid
```
Comment (cmyui) [inline common/ripple/user_utils.py]:
```suggestion
                JOIN {table_name} AS scores ON scores.id = scores_first.scoreid

```

File: common/ripple/user_utils.py [hunk 2]
```diff
@@ -951,36 +951,35 @@ async def recalculate_and_update_first_place_scores(user_id: int) -> None:
...
+                AND mode = %s
+                AND rx = %s
```
Comment (cmyui) [inline common/ripple/user_utils.py]:
```suggestion
                AND scores_first.mode = %s
                AND scores_first.rx = %s

```

File: common/ripple/user_utils.py [hunk 3]
```diff
@@ -951,36 +951,37 @@ async def recalculate_and_update_first_place_scores(user_id: int) -> None:
...
-                    table_name,
-                ),
-                [score["beatmap_md5"], score["play_mode"]],
+            existing_first_place = await glob.db.fetch(
+                f"""
+                SELECT scores_first.scoreid, scores_first.userid, scores.{order} AS score_value
+                FROM scores_first
+                WHERE beatmap_md5 = %s
+                JOIN {table_name} AS scores ON scores.id = scores_first.scoreid
+
```
Comment (cmyui) [inline common/ripple/user_utils.py]:
```suggestion

```

File: common/ripple/user_utils.py [hunk 4]
```diff
@@ -951,36 +951,36 @@ async def recalculate_and_update_first_place_scores(user_id: int) -> None:
...
-                [score["beatmap_md5"], score["play_mode"]],
+            existing_first_place = await glob.db.fetch(
+                f"""
+                SELECT scores_first.scoreid, scores_first.userid, scores.{order} AS score_value
+                FROM scores_first
+                WHERE beatmap_md5 = %s
+                JOIN {table_name} AS scores ON scores.id = scores_first.scoreid
+                AND scores_first.mode = %s
+                AND scores_first.rx = %s
+
```
Comment (cmyui) [inline common/ripple/user_utils.py]:
```suggestion
```

File: common/ripple/user_utils.py [hunk 5]
```diff
@@ -951,36 +951,35 @@ async def recalculate_and_update_first_place_scores(user_id: int) -> None:
...
+                """,
+                [score["beatmap_md5"], score["play_mode"], rx],
             )
 
             # Check if our score is better than the current #1.
             # If it is, then add/update scores_first.
             if (
-                not firstPlace
-                or score[order] > firstPlace[order]
-                or user_id == firstPlace["userid"]
```
Comment (cmyui) [inline common/ripple/user_utils.py:983]: I didn't understand this `user_id` check

**Review by tsunyoku — APPROVED** [review]

**Pull request merged**